Mathlib/Analysis/NormedSpace/LogarithmicDerivatives.lean

```diff
@@ -0,0 +1,130 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import Mathlib.Analysis.Complex.LocallyUniformLimit
+import Mathlib.Analysis.SpecialFunctions.Trigonometric.Deriv
+import Mathlib.Analysis.SpecialFunctions.Complex.LogDeriv
+
+/-!
+# Logarithmic Derivatives
+
+We define the logarithmic derivative of a function f as `deriv f / f`. We then prove some basic
+facts about this, including how it changes under multiplication and composition.
+
+We conclude by showing that the logarithmic derivative of a sequence of functions converging
+locally uniformly to a function is the logarithmic derivative of the limit function. This is useful
+for example for the Mittag-Leffler expansion of the cotangent function.
+-/
+
+noncomputable section
+
+open Filter Function
+
+open scoped Topology BigOperators Classical
+
+variable {𝕜 𝕜': Type*} [NontriviallyNormedField 𝕜] [NontriviallyNormedField 𝕜']
+  [NormedAlgebra 𝕜 𝕜']
+
+/-- The logarithmic derivative of a function defined as `deriv f /f`. Note that it will be zero
+at `x` if `f` is not DifferentiableAt `x`. -/
+def logDeriv (f : 𝕜 → 𝕜') :=
+  deriv f / f
+
+lemma logDeriv_eq_zero_of_not_differentiableAt (f : 𝕜 → 𝕜') (x : 𝕜) (h : ¬DifferentiableAt 𝕜 f x) :
+    logDeriv f x = 0 := by
+  simp only [logDeriv, Pi.div_apply, deriv_zero_of_not_differentiableAt h, zero_div]
+
+theorem logDeriv_id (x : 𝕜) : logDeriv id x = 1 / x := by
+  rw [logDeriv]
+  simp only [deriv_id', Pi.div_apply, id_eq, one_div]
+
+theorem logDeriv_const (a : 𝕜') : logDeriv (fun _ : 𝕜 ↦ a) = 0 := by
+  rw [logDeriv]
+  ext1 x
+  simp only [deriv_const', Pi.div_apply, zero_div, Pi.zero_apply]
+
+theorem LogDeriv_mul {f g : 𝕜 → 𝕜'} (x : 𝕜) (hfg : f x * g x ≠ 0) (hdf : DifferentiableAt 𝕜 f x)
+    (hdg : DifferentiableAt 𝕜 g x) :
+    logDeriv (fun z => f z * g z) x = logDeriv f x + logDeriv g x := by
+  simp only [logDeriv, Pi.div_apply, deriv_mul hdf hdg]
+  field_simp [(mul_ne_zero_iff.1 hfg).1, (mul_ne_zero_iff.1 hfg).2, mul_comm]
+
+theorem LogDeriv_mul_const {f : 𝕜 → 𝕜'} (x : 𝕜) (a : 𝕜') (hf : f x * a ≠ 0)
```

Comment:
It looks like you only need `a ≠ 0` for this lemma. If `f x = 0` or `f` is not differentiable at `x`, then both sides are zero.